engine/src/connector.rs

```diff
@@ -62,11 +61,15 @@ pub struct OnTransferMessageData {
     pub fee: U256,
 }
 
-impl<I: IO + Default + Copy> EthConnectorContract<I> {
+impl<I: IO + Copy> EthConnectorContract<I> {
```

Comment:
`Copy` (or at the very least `Clone`) is still required, unfortunately. The reason is because the engine and connector sub-components benefit from having a handle to the `IO` they can access in `self` methods (otherwise we end up passing in `io` all over the place and it's pretty messy). I think it might be ok if this ends up remaining the case in the long term, but perhaps in a future PR I'll see if I can sort out all the lifetimes to be able to pass a reference down instead of the actual object.